docs/posts/ibis-overturemaps/index.qmd

```diff
@@ -0,0 +1,213 @@
+---
+title: "From query to plot: Exploring GeoParquet Overture Maps with Ibis, DuckDB, and Lonboard"
+author: Naty Clementi and Kyle Barron
+date: 2024-09-13
+categories:
+  - blog
+  - duckdb
+  - overturemaps
+  - lonboard
+  - geospatial
+execute:
+  freeze: false
+---
+
+With the release of `DuckDB 1.1.0`, now we have support for reading GeoParquet
+files! With this exciting update we can query rich datasets from Overture Maps using python via Ibis with the performance of `DuckDB`.
+
+But the good news don't stop there, since `Ibis 9.2`, `lonboard` can plot data directly from an `Ibis` table, adding more simplicity and speed to your geospatial analysis.
+
+Let’s dive into how these tools come together.
+
+## Installation
+
+Install Ibis with the dependencies needed to work with geospatial data using DuckDB. To be able to read geoparquet files the duckdb version should be `>=1.1.0`.
+
+::: {.callout-note}
+At the moment duckdb 1.1.0 has a bug that prevents us from querying and writing the data to parquet using Ibis and DuckDB, so we are installing the `overturemaps` CLI to get the data.
+:::
+
+```bash
+$ pip install 'ibis-framework[duckdb,geospatial]' lonboard overturemaps
+```
+
+## Motivation
+
+Overture Maps offers a variety of datasets to query, but we thought that it would
+be interesting to see some plots related to the power infrastructure. We'll look
+into power plants, and power lines of most of the USA (excluding territories and
+Alaska for simplicity of the bounding box).
+
+## Download data
+
+**NOTE: not sure if I should have this here or avoid it because it's broken**
+
+With Ibis and DuckDB we could download only the necessary columns needed but at
+the moment this fails due to a bug.
+
+For future reference when this gets fixed, the expression would look like this:
+
+```python
+import ibis
+from ibis import _
+
+con = ibis.get_backend()
+
+# look into type infrastructure
+url = "s3://overturemaps-us-west-2/release/2024-07-22.0/theme=base/type=infrastructure/*"
+t = con.read_parquet(url, table_name="infra-usa")
+
+# filter for USA bounding box, subtype="power", and selecting only few columns
+expr = t.filter(_.bbox.xmin > -125.0, _.bbox.ymin > 24.8,
+                _.bbox.xmax < -65.8, _.bbox.ymax < 49.2,
+                _.subtype=="power"
+                ).select(["names",
+                          "geometry",
+                          "class",
+                          "sources",
+                          "source_tags"])
+
+```
+
+::: {.callout-note}
+If you inspect expr, you can see that the filters and projections get pushed down
+meaning you only bring the data that you asked for.
+:::
+
+```python
+# to write it to a parquet file we would do (currently broken)
+con.to_parquet(expr, "power-infra-usa.parquet")
+```
+
+But, no worries! Overture Maps has a nice [python CLI](https://docs.overturemaps.org/getting-data/overturemaps-py/) that allow us to download
```

Comment:
You can also do this from Python in case you wanted to avoid reaching for the command line, but this is fine.

Reply:
I just learned that the DuckDB bug fix will go out next week (see Max's comment below), so I'll rewrite this part to read download the data that way.